code/entities/ButtonEntity.cs

```diff
@@ -64,13 +64,16 @@ protected void SetOn( bool on )
 		{
 			RenderColor = On ? Color.Green : Color.Red;
 		}
-		this.WireTriggerOutput( "On", On );
-	}
 
+		var curOn = (bool)(this as IWireOutputEntity).GetOutput("On").value;
+		if (curOn != On)
```

Comment:
Your earlier effort to handle this in WireTriggerOutput should have covered this, but it would seem due to the late-type-binding we need to use `output.value.Equals( value )` rather than `output.value == (object)value` as you initially tried. I'll fix that!

With that resolved, I'm not sure this explicit check is as useful, since I think performance wise its basically identical to what WireTriggerOutput would be doing anyway.

Reply:
The issue without the check is that when I have a toggle input that should activate only when the input bool is true, holding a button would constantly be firing triggeroutput with true, and it would be constantly calling my input (in this case opening and closing the stargate iris). However it should not, since the button is not being repeatedly pressed when being held down, so this fixes that issue.